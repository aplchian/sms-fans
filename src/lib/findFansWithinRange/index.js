const zipcodes = require("zipcodes")
const { curry, filter } = require("ramda")
const { Maybe } = require("ramda-fantasy")

const findFansWithinRange = curry((distance, zipCode, fans) => {
  const fanInRange = fan => zipcodes.distance(fan.zip, zipCode) <= distance
  return Maybe(fans)
    .map(filter(fanInRange))
    .getOrElse([])
})

export default findFansWithinRange
