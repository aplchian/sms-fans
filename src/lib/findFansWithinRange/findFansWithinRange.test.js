import findFansWithinRange from "./"

test("find fans within range should work and return a fan", () => {
  const res = findFansWithinRange(100, "29405", getTestData())
  expect(res).toMatchSnapshot()
})

test("find fans within range should work and return no fans", () => {
  const res = findFansWithinRange(1, "29405", getTestData())
  expect(res).toMatchSnapshot()
})

function getTestData() {
  return [
    {
      zip: "29464",
      name: "alex"
    },
    {
      zip: "90210",
      name: "peter"
    }
  ]
}
