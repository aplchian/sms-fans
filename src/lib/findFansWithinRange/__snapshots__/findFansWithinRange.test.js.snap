// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`find fans within range should work and return a fan 1`] = `
Array [
  Object {
    "name": "alex",
    "zip": "29464",
  },
]
`;

exports[`find fans within range should work and return no fans 1`] = `Array []`;
